cypress/e2e/shared/notificationRules.test.ts

```diff
@@ -757,9 +757,426 @@ describe('NotificationRules', () => {
+    const rule2Clone = {
+      name: 'Despair',
+      description:
+        'Nabokov.  Hermann Karlovich, a Russian of German descent and owner of a chocolate factory, meets a homeless man in the city of Prague, who he believes is his doppelgänger.',
+      label: 'Chocolate',
+    }
+
+    const newName = 'The Gift'
+    const newEvery = '5m'
+    const newOffset = '1m'
+    const newLevel = 'WARN'
+
+    beforeEach(() => {
+      cy.writeLPDataFromFile({
+        filename: 'data/wumpus01.lp',
+        offset: '20m',
+        stagger: '1m',
+      })
+      cy.get<Organization>('@org').then((org: Organization) => {
+        cy.get<Bucket>('@bucket').then((bucket: Bucket) => {
+          // get default org and bucket
+          // 2. create check
+          const queryText = `from(bucket: \"${bucket.name}\")\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\n  |> filter(fn: (r) => r[\"_measurement\"] == \"wumpus\")\n  |> filter(fn: (r) => r[\"_field\"] == \"mag\")`
+          cy.createCheck({
+            ...check,
+            orgID: org.id,
+            query: {
+              ...check.query,
+              text: queryText,
+              builderConfig: {
+                ...check.query.builderConfig,
+                buckets: [bucket.id],
+              },
+            },
+          }).then((resp: {body: any}) => {
+            cy.wrap(resp.body).as('check')
+          })
+
+          cy.createEndpoint({
+            ...(endp as NotificationEndpoint),
+            orgID: org.id,
+          }).then(resp => {
+            cy.wrap(resp.body).as('endp')
+          })
+
+          cy.get<NotificationEndpoint>('@endp').then(endp => {
+            cy.createRule({
+              ...(rule as NotificationRule),
+              name: rule2Clone.name,
+              description: rule2Clone.description,
+              orgID: org.id,
+              endpointID: endp.id,
+            }).then(({body}) => {
+              cy.createAndAddLabel(
+                'notificationRules',
+                org.id,
+                body.id,
+                rule2Clone.label
+              )
+              cy.reload() // Alerts load from upper beforeEach might not catch new rule
+              cy.getByTestID('alerting-tab--rules').click()
+            })
+          })
+        })
+      })
+    })
+
+    it('can clone, modify and delete a notification rule', () => {
+      cy.log('clone rule')
+      cy.getByTestID(`rule-card ${rule2Clone.name}`).within(() => {
+        cy.getByTestID('context-menu-task').click()
+      })
+      cy.getByTestID('context-clone-task').click()
+
+      cy.getByTestID(`rule-card ${rule2Clone.name}`).within(() => {
+        cy.getByTestID('copy-task-id').then(elem => {
+          cy.wrap(
+            elem
+              .text()
+              .replace('Task ID: ', '')
+              .replace('Copy to Clipboard', '')
+          ).as('TargetTaskID')
+        })
+      })
+
+      // verify basic identifying parameters
```

Comment:
same pattern. cy.log() this comment.

Reply:
but it looks like repeated code (capturing IDs). can alternatively decide on helper functions.

Reply:
Added cy.log, but left off creating a helper function for capturing IDs. Did not see how a helper function would result in less code than the currently implementation using cypress/